server/ocr_server_signals.cpp

```diff
@@ -0,0 +1,50 @@
+#include <csignal>
+#include <vector>
+#include <stdlib.h>
+#include "lib/log.hpp"
+#include "ocr_server_signals.hpp"
+
+static void HandleInterruptSignal(int signum);
+static void HandleSegmentationFaultSignal(int signum);
+static void HandleHangupSignal(int signum);
+// TODO I don't like this module as it currently is, but it will do
+// for now. I want to be able to add function calls to a list of
+// called functions when a signal is fired.
+// e.x. SignalHandler.addSignalResponse(SIGHUP, someFunctionPointer);
+
+static signals::Signal all_signals[] = {
+  {.sig = SIGHUP, .func = HandleSegmentationFaultSignal},
```

Comment:
It looks like the signal handlers may be mixed up here, between hangup and segmentation fault.